xapian-bindings/android/CROSS_COMPILE

```diff
@@ -71,7 +71,7 @@ Save the file. Now generate the libxapian_jni.so by doing the following:
 
 export CXX=$TOOLCHAIN/bin/<host>-g++
 $CXX -c xapian_wrap.cc -I../../xapian-core/include
-$CXX -shared -o built/lib/libxapian_jni.so -Wl,-whole-archive libxapian-1.3.a  -Wl,-no-whole-archive xapian_wrap.o -lz -luuid -lc
+$CXX -shared -o built/lib/libxapian_jni.so -Wl,-whole-archive libxapian-x.x.a  -Wl,-no-whole-archive xapian_wrap.o -lz -luuid -lc
```

Comment:
For release versions (like 1.4.x), the library is just `libxapian.a`.  Only development versions have a suffix, and I think it makes more sense to make these instructions clearer for release versions.

But these steps (and the Java compilation below) seem to do by hand pretty much what's automated by `java/Makefile.am` - have you tried configuring xapian-bindings with something like `--with-java CXX=$TOOLCHAIN/bin/<host>-g++ JAVAC='javac -source 1.7 -target 1.7 -bootclasspath <path_to_jvm>/<your_jdk>/jre/lib/rt.jar'` ?

Was the issue libtool not managing to spot the static libxapian contained PIC code?